### PR TITLE
New package: ffmpeg6-6.0.1

### PR DESCRIPTION
**Pull request by zlice**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, **x86_64**, **i686**
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - **aarch64**
  - **x86_64-musl**

#### Comments

Makes `ffmpeg` a transitional meta package that is still version 4 but depends on `ffmpeg6` which has binaries and libraries. Should allow for a slow transition to move things over to 6 and avoid breaking current installs.

#### pre-reqs

- opencv https://github.com/void-linux/void-packages/pull/51124
- qtav remove https://github.com/void-linux/void-packages/pull/51123
- flowblade https://github.com/void-linux/void-packages/pull/50385
- idjc https://github.com/void-linux/void-packages/pull/48944

it-starts-timon

### CODE REVIEW

**Comment by classabbyamp** [comment]
- rebased
- made ffplay a transitional package too
- linted ffmpeg6 template

**Comment by classabbyamp** [comment]
not a fan of how `ffmpeg{,6}-devel` conflict with each other, but I think the only way to make that better is to add `replaces=`

**Review by ahesford — COMMENTED** [review]
I think we should eliminate all build options in the new `ffmpeg6` that aren't conditionally applied. By my count, that would leave the following options:
- `vaapi`
- `vdpau`
- `nvenc`
- `nvdec`
- `onevpl`
- `svtav1`

All other options that are currently part of `build_options_default` for all architectures can just be enabled unconditionally. In addition, I think `nvenc` and `nvdec` should be merged into one option, perhaps `nvcodec`, because there isn't much value in separating them.

File: srcpkgs/ffmpeg6/template [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+		libavformat6>=${version}_${revision}
+		libavutil6>=${version}_${revision}
+		libavfilter6>=${version}_${revision}
+		libpostproc6>=${version}_${revision}
+		libswscale6>=${version}_${revision}
+		libswresample6>=${version}_${revision}"
+	short_desc+=" - development files"
+	conflicts="ffmpeg-devel"
+	if [ "$XBPS_TARGET_MACHINE" = "i686" ]; then
+		# /usr/bin/strip: error: the input file '/destdir//ffmpeg-devel-4.4.4/usr/lib/libavfilter.a(vf_atadenoise.o)' has no sections
```
[the remark this answers is not in the document]
Comment (zlice) [reply srcpkgs/ffmpeg6/template:215]:
```
=> ffmpeg6-devel-6.0.1_1: running post-install hook: 06-strip-and-debug-pkgs ...
   Stripped static library: /usr/lib/libswresample.a
   Stripped static library: /usr/lib/libavcodec.a
   Stripped static library: /usr/lib/libavformat.a
   Stripped static library: /usr/lib/libswscale.a
   Stripped static library: /usr/lib/libavutil.a
   Stripped static library: /usr/lib/libpostproc.a
   Stripped static library: /usr/lib/libavdevice.a
/usr/bin/strip: error: the input file '/destdir//ffmpeg6-devel-6.0.1/usr/lib/libavfilter.a(vf_atadenoise.o)' has no sections
=> ERROR: ffmpeg6-devel-6.0.1_1: failed to strip /usr/lib/libavfilter.a
=> ERROR: ffmpeg6-devel-6.0.1_1: post-install_06-strip-and-debug-pkgs: 'find ${PKGDESTDIR} -type f' exited with 1
=> ERROR:   in hook() at common/hooks/post-install/06-strip-and-debug-pkgs.sh:68
=> ERROR:   in run_func() at common/xbps-src/shutils/common.sh:57
=> ERROR:   in run_pkg_hooks() at common/xbps-src/shutils/common.sh:314
=> ERROR:   in main() at common/xbps-src/libexec/xbps-src-doinstall.sh:66
```

forget the exact reason but other distros `--disable-stripping` and there was some ffmpeg/gcc/strip thing about the library not having sections bc 'x'

**Pull request closed without merging**